frontend/public/components/dashboard/details-card/detail-item.tsx

```diff
@@ -12,8 +15,28 @@ export const DetailItem: React.FC<DetailItemProps> = React.memo(({ title, value,
   </React.Fragment>
 ));
 
+
+export const DetailItemLink: React.FC<DetailItemLinkProps> = React.memo(({ title, value, link, isLoading }) => (
+  <React.Fragment>
+    <dt className="co-details-card__item-title">{title}</dt>
+    <dd className={classNames('co-details-card__item-value', {'text-secondary': !value})}>
+      {isLoading ? <LoadingInline /> : (
```

Comment:
would wait for  @cloudbehl @rawagner @gnehapk  perceptions on this custom details link ?

Reply:
nice, i'd like to use the same approach as we do for `InventoryItem` where we have a generic component, which takes isLoading, error and children https://github.com/openshift/console/blob/master/frontend/public/components/dashboard/inventory-card/inventory-item.tsx#L47.

In this case the child would be either string or `ExternalLink` or any other component if we need in the future. If error is true, then the unavailable is shown etc.

the basic component would be just `Item` and we will have `DetailItem` and `DetailItemExternalLink`

Reply:
Thanks @rawagner for pointing that out. However I will not be adding any required types to it and make error state detectable from the current parameters and also from the `error` parameter(optional) so that these changes do not break any existing references(However there are not many references and making changes to them is possible allowing us to make error a required parameter). Following your approach I don't think we will need to create a new component `DetailItemExternalLink` since we will be passing `ExternalLink` or other similar components as children.

Reply:
alright @rawagner , not aware of the implementation in `InventoryItem`.
Late to comment but I just did not find it appropriate and rather expecting something like that the `ExternalLink` to be passed  in `value`. Thank you!